File: arrays and hashing/Longest Consecutive Sequence.py
def longestConsecutive(nums: list[int]) -> int:
    seen = set(nums)

    ans = 0
    for num in nums:
        cur_ans = 1
        cur_num = num - 1
        while cur_num in seen:
            cur_ans += 1
            cur_num -= 1
        ans = max(ans, cur_ans)

    return ans


print(longestConsecutive([2,20,4,10,3,4,5]))
print(longestConsecutive([0,3,2,5,4,6,1,1]))

File: arrays and hashing/Longest Consecutive Sequence.cs
public class SolutionLongestConsecutive
{
    public int LongestConsecutive(int[] nums)
    {
        Dictionary<int, int> links = new Dictionary<int, int>();
        int ans = 0;

        foreach (int num in nums)
        {
            if (!links.ContainsKey(num))
            {
                int left = 0;
                int right = 0;
                if (links.ContainsKey(num + 1))
                    right = links[num + 1];
                if (links.ContainsKey(num - 1))
                    left = links[num - 1];

                links[num] = left + right + 1;
                links[num - left] = links[num];
                links[num + right] = links[num];
                ans = Math.Max(ans, links[num]);
            }
        }
        return ans;
    }
}
